[lcm] Respell LCM's external dependencies

Drake allows @glib to come from either pkg-config or bzlmod.
The latter offers BUILD files in each subdirectory, but the
former does not. We need to patch LCM to work with both.

Reasoning for not upstreaming this patch: Drake-specific
customization.

--- lcm/BUILD.bazel
+++ lcm/BUILD.bazel
@@ -143,7 +143,7 @@ LINKOPTS = select({
 
 DEPS = [
     ":_public_hdrs",
-    "@glib//glib",
+    "@glib",
 ]
 
 cc_library(

--- lcmgen/BUILD.bazel
+++ lcmgen/BUILD.bazel
@@ -28,6 +28,6 @@ cc_binary(
     visibility = ["//visibility:public"],
     deps = [
         "//lcm:lcm-version",
-        "@glib//glib",
+        "@glib",
     ],
 )

